managedtenants/utils/ocm.py

```diff
@@ -107,16 +103,18 @@ def add_addon(self, metadata):
         return self._post("/api/clusters_mgmt/v1/addons", json=addon)
 
     def _addon_exists(self, addon_id):
-        resp = self._get(f"/api/clusters_mgmt/v1/addons/{addon_id}")
-        return resp.status_code == 200
+        try:
+            self._get(f"/api/clusters_mgmt/v1/addons/{addon_id}")
+            return True
+        # `_get` raises OCMAPIError on 404's
+        except OCMAPIError:
+            return False
 
     def add_addon_version(self, imageset, metadata):
         if self._addon_exists(metadata.get("id")):
             return self._add_addon_version(imageset, metadata)
         # Create the addon first
-        resp = self.add_addon(metadata)
-        if resp.status_code != 201:
-            raise AddonCreationError("Failed to succesfully create the addon.")
+        self.add_addon(metadata)
```

Comment:
Let's handle some errors, for example, the way it's done here :

https://github.com/mt-sre/managed-tenants-cli/blob/415e8ab6a68e29e91f728bb9d4b2d757528f7729/managedtenants/utils/ocm.py#L159-L167

wdyt :)

Reply:
yep

Reply:
But there isn't any errors we can handle here, if the post call fails we just raise an exception. Its like how we do in other methods(where we just do ` return self._post("/api/clusters_mgmt/v1/addons", json=addon)`)

Reply:
Ohh sorted then, if it's meant to fail, then, I am happy :) (never thought I'd say this sentence while reviewing a PR :P )